.github/workflows/windows_noetic_build.yml

```
@@ -0,0 +1,38 @@
+name: Windows-Noetic-Build
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  schedule:
+    - cron: '0 5 * * *'
+
+jobs:
+  windows_ci:
+    name: Noetic
+    runs-on: windows-2019
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        path: src/plugin_loader
+    - name: Build
+      shell: cmd
+      run: |
+        choco sources add -n=roswin -s https://aka.ms/ros/public --priority 1
+
+        call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\Tools\VsDevCmd.bat" -arch=amd64 -host_arch=amd64
+
+        python -m pip install vcstool -q
+        python -m pip install colcon-common-extensions -q
+        choco install boost -y --no-progress
+
+        call RefreshEnv.cmd
+
+        vcs import --input "${{ github.workspace }}/src/plugin_loader/dependencies.repos" src/
+
+        colcon build --cmake-args -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=ON
+
+        colcon test
+
+        colcon test-result
```

Actually we should move this into a separate stage

```suggestion
        - name: Test
          shell: cmd
          run: |
            colcon test
            colcon test-result
```